### PR TITLE
Add core support with interop for Segmentation SDK. 

### PR DESCRIPTION
**Pull request by dmandar**
Add core support with interop for Segmentation SDK. Also update header file location to within sources folder.

### CODE REVIEW

File: Firebase/Core/FIRErrors.m [hunk 1]
```diff
@@ -19,3 +19,4 @@
 NSString *const kFirebaseCoreErrorDomain = @"com.firebase.core";
 NSString *const kFirebasePerfErrorDomain = @"com.firebase.perf";
 NSString *const kFirebaseStorageErrorDomain = @"com.firebase.storage";
+NSString *const kFirebaseSegmentationErrorDomain = @"com.firebase.segmentation";
```
Comment (paulb777) [inline Firebase/Core/FIRErrors.m]: Delete.  FIRErrors is deprecated and nothing new should be added.

File: Firebase/Core/Private/FIRErrors.h [hunk 2]
```diff
@@ -22,3 +22,4 @@ extern NSString *const kFirebaseErrorDomain;
 extern NSString *const kFirebaseConfigErrorDomain;
 extern NSString *const kFirebaseCoreErrorDomain;
 extern NSString *const kFirebasePerfErrorDomain;
+extern NSString *const kFirebaseSegmentationErrorDomain;
```
Comment (paulb777) [inline Firebase/Core/Private/FIRErrors.h]: Delete. Same as above.

File: FirebaseSegmentation/Sources/FIRSegmentation.m [hunk 3]
```diff
@@ -12,4 +12,50 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#import "FIRSegmentation.h"
+#import "FirebaseSegmentation/Sources/Public/FIRSegmentation.h"
+
+#import "FIRAppInternal.h"
```
Comment (paulb777) [inline FirebaseSegmentation/Sources/FIRSegmentation.m]: Should be #import <FirebaseCore/FIRAppInternal.h> for this and other Core Private headers.

File: FirebaseSegmentation/Sources/FIRSegmentationComponent.m [hunk 4]
```diff
@@ -0,0 +1,86 @@
+#import "FIRSegmentationComponent.h"
+
+#import "FIRAppInternal.h"
```
Comment (paulb777) [inline FirebaseSegmentation/Sources/FIRSegmentationComponent.m]: Use module import format here too.

File: FirebaseSegmentation/Sources/Private/FIRSegmentationComponent.h [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+/// Cached instances of Segmentation objects.
+@property(nonatomic, strong) NSMutableDictionary<NSString *, FIRSegmentation *> *instances;
+
+/// Default method for retrieving a Segmentation instance, or creating one if it doesn't exist.
+- (FIRSegmentation *)segmentation;
+
+@end
+
+/// A concrete implementation for FIRSegmentationInterop to create Segmentation instances and
```
Comment (paulb777) [inline FirebaseSegmentation/Sources/Private/FIRSegmentationComponent.h]: Is there a need for FIRSegmentationInterop?

File: FirebaseSegmentation.podspec [hunk 6]
```diff
@@ -21,7 +21,7 @@ Firebase Segmentation enables you to associate your custom application instance
   s.prefix_header_file = false
 
   s.source_files = 'FirebaseSegmentation/Sources/**/*'
-  s.public_header_files = 'FirebaseSegmentation/Public/*.h'
+  s.public_header_files = 'FirebaseSegmentation/Sources/Public/*.h'
 
   s.dependency 'FirebaseCore', '~> 6.0'
```
Comment (paulb777) [inline FirebaseSegmentation.podspec]: 6.1

File: Firebase/Core/FIRErrors.m (hunk 1 above)
Comment (dmandar) [reply Firebase/Core/FIRErrors.m]: Done.

File: Firebase/Core/Private/FIRErrors.h (hunk 2 above)
Comment (dmandar) [reply Firebase/Core/Private/FIRErrors.h]: Done.

File: FirebaseSegmentation/Sources/FIRSegmentation.m (hunk 3 above)
Comment (dmandar) [reply FirebaseSegmentation/Sources/FIRSegmentation.m]: Done.

File: FirebaseSegmentation/Sources/FIRSegmentationComponent.m (hunk 4 above)
Comment (dmandar) [reply FirebaseSegmentation/Sources/FIRSegmentationComponent.m]: Done.

File: FirebaseSegmentation/Sources/Private/FIRSegmentationComponent.h (hunk 5 above)
Comment (dmandar) [reply FirebaseSegmentation/Sources/Private/FIRSegmentationComponent.h]: Is there a reason not to? What is the alternative? I thought all products were moving to using interop.

File: FirebaseSegmentation.podspec (hunk 6 above)
Comment (dmandar) [reply FirebaseSegmentation.podspec]: Done.

File: FirebaseSegmentation/Sources/Private/FIRSegmentationComponent.h (hunk 5 above)
Comment (paulb777) [reply FirebaseSegmentation/Sources/Private/FIRSegmentationComponent.h]: Interop is only needed for products that want to manage their interface to other Firebase dependencies separately from their library implementation.  Will Segmentation have Firebase pod dependencies?

**Comment by paulb777** [comment]
Please fix the travis failure before merging.

File: FirebaseSegmentation/Sources/FIRSegmentationComponent.m [hunk 7]
```diff
@@ -0,0 +1,89 @@
+#import "FIRSegmentationComponent.h"
+
+#import <FirebaseCore/FIRAppInternal.h>
+#import <FirebaseCore/FIRComponentContainer.h>
+#import <FirebaseCore/FIROptionsInternal.h>
+#import <FirebaseSegmentation/Sources/Private/FIRSegmentationInternal.h>
```
[the remark this answers is not in the document]
Comment (dmandar) [reply FirebaseSegmentation/Sources/FIRSegmentationComponent.m]: Done.

**Comment by dmandar** [comment]
Yes, segmentation will have IID and FIS as pod dependencies.

File: FirebaseSegmentation/Sources/Private/FIRSegmentationComponent.h (hunk 5 above)
Comment (dmandar) [reply FirebaseSegmentation/Sources/Private/FIRSegmentationComponent.h]: Yes, Segmentation will need to depend on IID and FIS.

**Pull request merged**